firmware/arduino/chessboard/uart.ino

```diff
@@ -1,227 +1,225 @@
-        }
-        // Add byte to buffer
-        else if (byteNum  !=  -1)
-        {
-            rxBufferPtr[byteNum++] = incomingByte;
-        }
-
-        // Check if the buffer is full, process input
-        if (byteNum  ==  INCOMING_MESSAGE_LENGTH)
-        {
-            // Reset buffer position
-            byteNum = -1;
-
-            // Swap rxBufferPtr and receivedMessagePtr pointers
-            tempCharPtr = rxBufferPtr;
-            rxBufferPtr = receivedMessagePtr;
-            receivedMessagePtr = tempCharPtr;
-
-            // Tell game loop to process input
-            receivedMessageValidFlag = true;
-        }
+      // Send message to Pi if the previous instruction was incomplete
+      if (byteNum != -1)
+      {
+        currentState = ERROR;
+        extraByte = INCOMPLETE_INSTRUCTION;
+        uartMessageIncompleteFlag = true;
+      }
+
+      byteNum = 0;
+    }
+    // Add byte to buffer
+    else if (byteNum  !=  -1)
+    {
+      rxBufferPtr[byteNum++] = incomingByte;
+    }
+
+    // Check if the buffer is full, process input
+    if (byteNum  ==  INCOMING_MESSAGE_LENGTH)
+    {
+      // Reset buffer position
+      byteNum = -1;
+
+      // Swap rxBufferPtr and receivedMessagePtr pointers
+      tempCharPtr = rxBufferPtr;
+      rxBufferPtr = receivedMessagePtr;
+      receivedMessagePtr = tempCharPtr;
+
+      // Tell game loop to process input
+      receivedMessageValidFlag = true;
     }
+  }
 }
 
 // Check that the instruction is valid
 bool validateMessageFromPi(volatile char * message)
 {
-    // If no error occurs to change the extraByte, it should store the opcode
-    extraByte = message[0];
+  // If no error occurs to change the extraByte, it should store the opcode
+  extraByte = message[0];
 
-    if (message[0] == DIRECT || message[0] == EDGES)
+  if (message[0] == DIRECT || message[0] == EDGES)
+  {
+    if (isInvalidCoord(message[1]) || isInvalidCoord(message[2]) ||
+        isInvalidCoord(message[3]) || isInvalidCoord(message[4]))
     {
-        if (isInvalidCoord(message[1]) || isInvalidCoord(message[2]) ||
-            isInvalidCoord(message[3]) || isInvalidCoord(message[4]))
-        {
-            extraByte = INVALID_LOCATION;
-            currentState = ERROR;
-            return false;
-        }
+      extraByte = INVALID_LOCATION;
+      currentState = ERROR;
+      return false;
     }
-    else if (message[0] == ALIGN)
+  }
+  else if (message[0] == ALIGN)
+  {
+    if (isInvalidCoord(message[1]) || isInvalidCoord(message[2]))
     {
-        if (isInvalidCoord(message[1]) || isInvalidCoord(message[2]))
-        {
-            extraByte = INVALID_LOCATION;
-            currentState = ERROR;
-            return false;
-        }
+      extraByte = INVALID_LOCATION;
+      currentState = ERROR;
+      return false;
     }
-    else if (message[0] == INSTRUCTION)
+  }
+  else if (message[0] == INSTRUCTION)
+  {
+    // Check if message[5] holds an invalid instruction type
```

Comment:
Should be `Check if message[1] holds an invalid instruction type or message[2] is an invalid code`.